pandas/tests/series/indexing/test_mask.py

```diff
@@ -67,3 +69,26 @@ def test_mask_inplace():
     rs = s.copy()
     rs.mask(cond, -s, inplace=True)
     tm.assert_series_equal(rs, s.mask(cond, -s))
+
+
+@pytest.mark.parametrize(
+    "dtype",
+    [
+        "Int64",
+        pytest.param("int64[pyarrow]", marks=td.skip_if_no("pyarrow")),
+    ],
+)
+def test_mask_na(dtype):
+    # We should not be filling pd.NA. See GH#60729
+    series = Series([None, 1, 2, None, 3, 4, None], dtype=dtype)
+    cond = series <= 2
+    expected = Series([None, -99, -99, None, 3, 4, None], dtype=dtype)
+
+    result = series.mask(cond, -99)
+    tm.assert_series_equal(result, expected)
+
+    result = series.mask(cond.to_list(), -99)
+    tm.assert_series_equal(result, expected)
+
+    result = series.mask(cond.to_numpy(), -99)
+    tm.assert_series_equal(result, expected)
```

Comment:
Can you instead parametrize this test. Something like:

    @pytest.mark.parametrize("cond_type", ["series", "list", "numpy"])

and

```python
if cond_type == "list":
    cond = cond.to_list()
elif cond_type == "numpy":
    cond = cond.to_numpy()
```